tests/pages/firefox/developer.py

```diff
@@ -28,14 +27,14 @@ def is_secondary_download_button_displayed(self):
 
     @property
     def developer_videos(self):
-        return [Video(self.selenium, root=el) for el in
-                self.selenium.find_elements(*self._videos_locator)]
+        return [Video(self.base_url, self.selenium, root=el) for el in
+                self.find_elements(self._videos_locator)]
 
 
 class Video(PageRegion):
 
     def play(self):
-        modal = Modal(self.selenium)
-        self.root.click()
-        Wait(self.selenium, self.timeout).until(lambda s: modal.is_displayed)
+        modal = Modal(self.base_url, self.selenium)
+        self._root.click()
```

Comment:
I'm wondering why this is just about the only place we need to refer to `self._root` in a page object. What's special about this and can we avoid it?